### PR TITLE
Initial checkin of a multicorn-based FDW

### PR DESCRIPTION
**Pull request by lorton**

### CODE REVIEW

**Comment by lorton** [comment]
Any thoughts on this @greglandrum or @cdvonbargen ?

**Comment by greglandrum** [comment]
Apologies that I haven’t looked at this yet. I still haven’t managed to get a GPU similarity server running (the GPU in my Linux box is too old and I haven’t put in the work to get a non-GPU version working). I am away from the computer this weekend, but will make some time on Monday

File: fdw/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+             smiles              |      id       | similarity
+---------------------------------+---------------+------------
+ CCCC                            | CHEMBL134702  |          1
+ CCCCC                           | ZINC01698513  |        0.5
+ CCCCCC                          | ZINC01532209  |        0.5
+ CCCCC                           | CHEMBL16102   |        0.5
+ CCCCCC                          | CHEMBL15939   |        0.5
+ CCC                             | CHEMBL135416  |        0.5
+......
+```
```
Comment (greglandrum) [inline fdw/README.md]: Probably worth adding something here about how the query can be changed without dropping/recreating the foreign table by doing:
```ALTER FOREIGN TABLE table similarity_search options (SET smiles 'COc1ccc2cc(C(C)C(=O)O)ccc2c1');```

**Comment by greglandrum** [comment]
This was trickier to get working than I had hoped, but that was because getting multicorn to install took a bit (which was user error on my part). Once installed everything seems to work well.

The code is simple and straightforward. LGTM

File: fdw/README.md (hunk 1 above)
Comment (lorton) [reply fdw/README.md]: Ah yeah - that's a good suggestion.  My one concern with this interface is the lack of multiuser/thread safety.  I suppose users would always have to be using their own unique foreign table names to make sure they're not stomping on each others feet.  That said, I don't think we have any other option for FDW.  Do you have any thoughts about whether the interface itself is reasonable?

**Comment by greglandrum** [comment]
I have a suggestion for how to provide the query SMILES without needing to modify the foreign table or its options. It feels like a bit of a hack but it sure does make using the FDW a ton easier:
```select smiles,id,similarity from similarity_search where query='COc1ccc([C@H](C)C(=O)OC)cc1';```
feels a lot better than:
```
ALTER FOREIGN TABLE table similarity_search options (SET smiles 'COc1ccc2cc(C(C)C(=O)O)ccc2c1');
select * from similarity_search;
```
How do you want me to share that? Should I provide a diff or do a PR against the `fdw` branch?

File: fdw/README.md (hunk 1 above)
Comment (greglandrum) [reply fdw/README.md]: Agreed that this is ugly and should probably be ignored. The alternate approach I mention in the later comment solves this problem.

Comment (lorton) [reply fdw/README.md]: Hi @greglandrum - that sounds like an improvement for sure, feel free to push against this branch to this PR.

Comment (greglandrum) [reply fdw/README.md]: ok, it's #24

File: fdw/queries.sql.example [hunk 2]
```diff
@@ -0,0 +1,12 @@
+drop foreign table similarity_search;
```
Comment (greglandrum) [inline fdw/queries.sql.example]: I'm not sure what this file is for, but if it needs to be here, this first line should be:
`drop foreign table if exists similarity_search;`
and the use of `SMILES` below (line 10) should be removed.

**Comment by greglandrum** [comment]
LGTM

**Pull request merged**